src/main/java/net/ripe/ipresource/jdk17/Ipv6Address.java

```diff
@@ -34,19 +34,25 @@
 import org.jetbrains.annotations.NotNull;
 
 import java.math.BigInteger;
+import java.nio.ByteBuffer;
+import java.util.regex.Pattern;
 
 public final class Ipv6Address implements IpAddress {
     public static final int NUMBER_OF_BITS = 128;
 
     public static final Ipv6Address LOWEST = new Ipv6Address(0, 0);
     public static final Ipv6Address HIGHEST = new Ipv6Address(-1, -1);
 
+    /* Pattern to match IPv6 addresses in forms defined in http://www.ietf.org/rfc/rfc4291.txt */
```

Comment:
It looks quite intimidating and complex.
Don't we want to use some library for parsing?
https://github.com/AdoptOpenJDK/openjdk-jdk11/blob/master/src/java.base/share/classes/sun/net/util/IPAddressUtil.java
Or Guava or something